### ISSUE TITLE
Issue with loading tiles in Starbound

### ISSUE THREAD

**Issue by negezor** [OpenStarbound/OpenStarbound#60]
One of the most annoying things in Starbound for me remains the game rendering on the processor and tile loading during movement.

A demonstration of what I mean can be seen here: https://youtu.be/IqV3Nl6AIc4. This even occurs during simple planet traversal, which greatly detracts from the overall experience.

I really wish this could be fixed. Has anyone conducted profiling to understand why this is happening? Is this a fundamental issue with the network code? Or is it only a local client problem?

**Comment by Kilkenni** [comment]
An interesting question and an old issue indeed.
It's a pure guess, but I'd say it's a server-side problem. I suspect that chunks of the world are being generated on-the-fly (since the planet normally is not generated entirely).
SB, even in single mode, works on client-server model, and I assume that proc gen is handled by the server.

**Comment by negezor** [comment]
Based on that assumption, perhaps it might be possible to increase the size of the tiles sent from the server by +20% of the current View to minimize this? :thinking:

**Comment by negezor** [comment]
It's a singleplayer. On the server, the problem is more obvious. What's odd is that it was less pronounced on common Starbound rather than OpenStarbound. 
```
/run return root.assetJson("/client.config:windowMonitoringBorder")
32
```

If it helps here is my list of mods:
- https://steamcommunity.com/sharedfiles/filedetails/?id=729480149
- https://steamcommunity.com/sharedfiles/filedetails/?id=2017503832
- https://steamcommunity.com/sharedfiles/filedetails/?id=2512589532
- https://steamcommunity.com/sharedfiles/filedetails/?id=1933871933
- https://steamcommunity.com/sharedfiles/filedetails/?id=816649310
- https://steamcommunity.com/sharedfiles/filedetails/?id=1194878261
- https://steamcommunity.com/sharedfiles/filedetails/?id=1307942879
- https://steamcommunity.com/sharedfiles/filedetails/?id=729599895
- https://steamcommunity.com/sharedfiles/filedetails/?id=729558042
- https://steamcommunity.com/sharedfiles/filedetails/?id=729426722
- https://steamcommunity.com/sharedfiles/filedetails/?id=1642552023
- https://steamcommunity.com/sharedfiles/filedetails/?id=729492703
- https://steamcommunity.com/sharedfiles/filedetails/?id=761242493
- https://steamcommunity.com/sharedfiles/filedetails/?id=729427606

**Comment by Kilkenni** [comment]
> hi - what is your latency to the server in the video (or is it single-player)? also, by default Starbound pads the client's chunk loading region by 32 tiles. has this value been changed by any mods you're using? (you can check by typing `/run return root.assetJson("/client.config:windowMonitoringBorder")` in the chat)

I wonder if it's a good idea to increase this padding for OpenSb. Or! Make it a setting customizable through game options, akin to "loading distance" for 3D games. I mean, theoretically modern computers should easily handle more than 32 tiles.

I wonder, however, how will it impact gameplay. If this is an option, there can be a case where a client has lower "padding" than the server, meaning that server should probably know about client's "padding distance" to supply client with appropriately "cropped" data as compared to what the server processes.

In other words, a server will be able to simulate a large area but send to client only a cropped part of this region depending on client settings.
